market/listeners.py

```diff
@@ -0,0 +1,34 @@
+__author__ = 'chris'
+import json
+from interfaces import MessageListener
+from zope.interface import implements
+from db.datastore import MessageStore
+from twisted.internet import task, reactor
+
+class MessageListenerImpl(object):
+    implements(MessageListener)
+
+    def __init__(self, web_socket_factory):
+        self.ws = web_socket_factory
+        self.db = MessageStore()
+
+    def notify(self, sender_guid, signed_pubkey, encryption_pubkey,
+               subject, message_type, message, timestamp, signature):
+
+        self.db.save_message(sender_guid, signed_pubkey, encryption_pubkey, subject,
```

Comment:
Maybe we should make a Message object rather than pass in an insane amount of parameters?